core/modules/link/link.module

```diff
@@ -930,6 +940,54 @@ function link_field_process($element, $form_state, $complete_form) {
   return $element;
 }
 
+/**
+ * Link field element validation.
+ */
+function link_field_element_validate(&$element, &$form_state, $complete_form) {
+  $value = $element['#value'];
+  $is_default_field_value = FALSE;
+
+  // Special validations when the values are not the default and when url has a
+  // value. NB: validation for subelements marked as 'required' is done by core.
+  if (!empty($value['url']) && !$is_default_field_value) {
```

Comment:
`$is_defaul_field_value` is always FALSE, and the negation is always TRUE, therefore this is not needed. You can remove the variable `$is_defaul_field_value` completely. This was here before, when the validation checked to see if the value of the element matched the default value. I don't think this is necessary anymore.